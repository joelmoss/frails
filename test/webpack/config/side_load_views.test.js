const createConfig = require('../../../package/webpack/create_config')

describe('config.sideLoadViews', () => {
  test('boolean', () => {
    expect(createConfig({ sideLoadViews: false })).toMatchSnapshot()
    expect(createConfig({ sideLoadViews: true })).toMatchSnapshot()
  })

  test('existing entry prop', () => {
    expect(createConfig({ entry: { 'some/path': 'some/path.js' } }).entry).toMatchSnapshot()
  })
})
