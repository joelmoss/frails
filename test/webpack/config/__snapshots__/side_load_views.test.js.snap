// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config.sideLoadViews boolean 1`] = `
Object {
  "devServer": Object {
    "contentBase": "<PROJECT_ROOT>/public/assets",
    "port": 8080,
    "public": "localhost:8080",
    "publicPath": "/assets/",
    "watchOptions": Object {
      "ignored": "**/node_modules/**",
    },
  },
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "alias": Object {
      "assets": "<PROJECT_ROOT>/app/assets",
    },
  },
}
`;

exports[`config.sideLoadViews boolean 2`] = `
Object {
  "devServer": Object {
    "contentBase": "<PROJECT_ROOT>/public/assets",
    "port": 8080,
    "public": "localhost:8080",
    "publicPath": "/assets/",
    "watchOptions": Object {
      "ignored": "**/node_modules/**",
    },
  },
  "entry": Object {
    "views/layouts/application": Array [
      "<PROJECT_ROOT>/app/views/layouts/application.css",
      "<PROJECT_ROOT>/app/views/layouts/application.js",
    ],
    "views/layouts/no_side": Array [
      "<PROJECT_ROOT>/app/views/layouts/no_side.css",
    ],
    "views/pages/_header": Array [
      "<PROJECT_ROOT>/app/views/pages/_header.css",
    ],
    "views/pages/_post": Array [
      "<PROJECT_ROOT>/app/views/pages/_post.css",
    ],
    "views/pages/home": Array [
      "<PROJECT_ROOT>/app/views/pages/home.css",
    ],
  },
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "alias": Object {
      "assets": "<PROJECT_ROOT>/app/assets",
    },
  },
}
`;

exports[`config.sideLoadViews existing entry prop 1`] = `
Object {
  "some/path": "some/path.js",
  "views/layouts/application": Array [
    "<PROJECT_ROOT>/app/views/layouts/application.css",
    "<PROJECT_ROOT>/app/views/layouts/application.js",
  ],
  "views/layouts/no_side": Array [
    "<PROJECT_ROOT>/app/views/layouts/no_side.css",
  ],
  "views/pages/_header": Array [
    "<PROJECT_ROOT>/app/views/pages/_header.css",
  ],
  "views/pages/_post": Array [
    "<PROJECT_ROOT>/app/views/pages/_post.css",
  ],
  "views/pages/home": Array [
    "<PROJECT_ROOT>/app/views/pages/home.css",
  ],
}
`;
